pkg/core/resources/apis/mesh/validators.go

```diff
@@ -364,16 +364,24 @@ func ValidateTargetRef(
 		}
 		err.Add(disallowedField("mesh", ref.Mesh, ref.Kind))
 		err.Add(disallowedField("tags", ref.Tags, ref.Kind))
+		err.Add(disallowedField("labels", ref.Labels, ref.Kind))
+		err.Add(disallowedField("namespace", ref.Namespace, ref.Kind))
+		err.Add(disallowedField("sectionName", ref.SectionName, ref.Kind))
 	case common_api.MeshSubset:
 		err.Add(disallowedField("name", ref.Name, ref.Kind))
 		err.Add(disallowedField("mesh", ref.Mesh, ref.Kind))
 		err.Add(ValidateTags(validators.RootedAt("tags"), ref.Tags, ValidateTagsOpts{}))
+		err.Add(disallowedField("labels", ref.Labels, ref.Kind))
+		err.Add(disallowedField("namespace", ref.Namespace, ref.Kind))
+		err.Add(disallowedField("sectionName", ref.SectionName, ref.Kind))
 	case common_api.MeshService, common_api.MeshHTTPRoute:
-		err.Add(requiredField("name", ref.Name, ref.Kind))
 		err.Add(validateName(ref.Name, opts.AllowedInvalidNames))
 		err.Add(disallowedField("mesh", ref.Mesh, ref.Kind))
 		err.Add(disallowedField("tags", ref.Tags, ref.Kind))
 		err.Add(disallowedField("proxyTypes", ref.ProxyTypes, ref.Kind))
+		if len(ref.Labels) > 0 && (ref.Name != "" || ref.Namespace != "") {
```

Comment:
things I am not 100% sure:
- should we be able to combine `sectionName` with labels? (from implementation point of view section name will be just another tag, so maybe you can specify it as label?)
- should we be able to specify targetRef with just `sectionName`?
- should we have at least name or namespace if labels are empty? having empty targetRef makes no sense, so we should probably have at least one of thm

Reply:
1. Probably we can just restrict this for now and open it up if there's a use case
2. You mean without a name? No IMO 
3. Yes I think if there's no labels it requires a name and if there's no namespace it's the current namespace

Reply:
I've updated validation accordingly to the comment